tests/pc/test_lag_member.py

```diff
@@ -0,0 +1,360 @@
+import pytest
+
+import time
+import logging
+import json
+
+from tests.common.helpers.assertions import pytest_assert, pytest_require
+from tests.ptf_runner import ptf_runner
+from tests.common.fixtures.ptfhost_utils import copy_ptftests_directory   # lgtm[py/unused-import]
+from tests.common.config_reload import config_reload
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology("t0")
+]
+
+PTF_LAG_NAME = "bond1"
+DUT_LAG_NAME = "PortChannel1"
+ATTR_PORT_NOT_BEHIND_LAG = "port_not_behind_lag"
+ATTR_PORT_BEHIND_LAG = "port_behind_lag"
+TEST_DIR = "/tmp/acstests/"
+HWSKU_INTF_NUMBERS_DICT = {
+    "Mellanox-SN2700": 16,
+    "ACS-MSN4600C": 16,
+    "Mellanox-SN2700": 16,
+    "Mellanox-SN3800-D112C8": 16,
+    "Force10-S6000": 16
+}
+
+def add_member_to_vlan(duthost, vlan_id, member_name, params):
+    """
+    Add vlan member
+
+    Args:
+        duthost: DUT host object
+        vlan_id: id of vlan
+        member_name: interface added to vlan
+        params: additional params in command line
+    """
+    duthost.shell("config vlan member add {} {} {}".format(params, vlan_id, member_name))
+
+def remove_member_from_vlan(duthost, vlan_id, member_name):
+    """
+    Remove vlan member
+
+    Args:
+        duthost: DUT host object
+        vlan_id: id of vlan
+        member_name: interface removed from vlan
+    """
+    duthost.shell("config vlan member del {} {}".format(vlan_id, member_name))
+
+def restart_ptf_nn_agent(ptfhost):
+    """
+    Restart ptf_nn_agent
+
+    Args:
+        ptfhost: PTF host object
+    """
+    ptfhost.shell("supervisorctl restart ptf_nn_agent")
+
+def remove_ip_from_port(duthost, port, ip=None):
+    """
+    Remove ip addresses from port
+
+    Args:
+        duthost: DUT host object
+        port: port name
+        ip: IP address
+    """
+    ip_addresses = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"].get("INTERFACE", {}).get(port, {})
+    if ip_addresses:
+        for ip in ip_addresses:
+            duthost.shell("config interface ip remove {} {}".format(port, ip))
+    elif ip:
+        duthost.shell("config interface ip remove {} {}".format(port, ip))
+
+def dut_teardown(duthost, dut_lag_map):
+    """
+    Restore dut configuration
+
+    Args:
+        duthost: DUT host object
+        dut_lag_map: imformation about lag in dut
+    """
+    try:
+        for dut_lag_member in dut_lag_map[DUT_LAG_NAME]:
+            duthost.shell("config portchannel member del {} {}".format(DUT_LAG_NAME, dut_lag_member))
+        add_member_to_vlan(duthost, 1000, dut_lag_member, "-u")
+
+        remove_member_from_vlan(duthost, dut_lag_map["vlan"]["id"], DUT_LAG_NAME)
+        duthost.shell("config portchannel del {}".format(DUT_LAG_NAME))
+        remove_member_from_vlan(duthost, dut_lag_map["vlan"]["id"], dut_lag_map[ATTR_PORT_NOT_BEHIND_LAG]["port_name"])
```

Comment:
you can also extract a method for these steps, such as `move_vlan_port_to_another_vlan()`
and also considering if we have any chance to increase the readability
`move_vlan_port_to_pc` ...

Reply:
Done